libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1051,6 +1051,20 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 		);
 		break;
 	}
+	case FunctionType::Kind::Wrap:
+	case FunctionType::Kind::Unwrap:
+	{
+		solAssert(arguments.size() == 1, "");
+		FunctionType::Kind kind = functionType->kind();
+		auto& userDefinedType = kind == FunctionType::Kind::Wrap ?
+			dynamic_cast<UserDefinedValueType const&>(type(_functionCall)) :
+			dynamic_cast<UserDefinedValueType const&>(type(*arguments.at(0)));
+		if (kind == FunctionType::Kind::Wrap)
+			solAssert(type(*arguments.at(0)).isImplicitlyConvertibleTo(userDefinedType.underlyingType()), "");
```

Comment:
Not sure this is necessary, but if we want it, then

```suggestion
		if (kind == FunctionType::Kind::Wrap)
			solAssert(type(*arguments.at(0)).isImplicitlyConvertibleTo(
				dynamic_cast<UserDefinedValueType const&>(type(_functionCall))
			);
```
would be enough.